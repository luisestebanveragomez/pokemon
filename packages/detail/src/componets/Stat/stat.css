
.stat__table {
  display: flex;
  gap: 1rem;
  color: var(--white);
  margin-bottom: 10px;
}

.stat__table-name {
  width: 120px;
  text-transform: capitalize;
}

.stat__table-bar {
  flex: 1;
}

.stat__table-bar div {
  height: 100%;
  background-color: var(--white);
  border-radius: 10px;
  width: 0;
  transition: .3s;
}

.stat__table-base {
  width: 30px;
}