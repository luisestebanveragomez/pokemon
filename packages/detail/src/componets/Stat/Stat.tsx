import { FC, useEffect, useState } from "react";

import "./stat.css";

interface StatProps {
  name: string;
  percentage: number;
  base: number;
}

export const Stat: FC<StatProps> = ({ name, percentage, base }) => {
  const [styles, setStyles] = useState({});

  useEffect(() => {
    setTimeout(
      () =>
        setStyles({
          width: `${percentage}%`,
        }),
      700
    );
  }, [percentage]);

  return (
    <section className="stat__table">
      <div className="stat__table-name">{name}</div>
      <div className="stat__table-bar">
        <div style={styles}></div>
      </div>
      <div className="stat__table-base">{base}</div>
    </section>
  );
};
