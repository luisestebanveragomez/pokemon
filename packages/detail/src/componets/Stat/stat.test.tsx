import { render, waitFor } from '@testing-library/react';
import { Stat } from './Stat';
import { act } from 'react-dom/test-utils';

describe('Stat Component', () => {

  it('Should Be Render', async () => {
    const {container} = render(<Stat name='test' base={12} percentage={20}/>)
    expect(container).toMatchSnapshot()
  })

  it('Should Be Validate Value', async () => {
    jest.useFakeTimers();
    const {container} = render(<Stat name='test' base={12} percentage={20}/>)
    act(() => {
      jest.runAllTimers();
    });
    const el = document.querySelector('.stat__table-name')
    expect(el.textContent).toEqual('test')
  })
})