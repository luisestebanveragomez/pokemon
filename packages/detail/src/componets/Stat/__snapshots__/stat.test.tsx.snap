// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat Component Should Be Render 1`] = `
<div>
  <section
    class="stat__table"
  >
    <div
      class="stat__table-name"
    >
      test
    </div>
    <div
      class="stat__table-bar"
    >
      <div />
    </div>
    <div
      class="stat__table-base"
    >
      12
    </div>
  </section>
</div>
`;
